src/ProgressOnderwijsUtils/Extensions/TreeExtensions.cs

```diff
@@ -64,5 +66,73 @@ public static IEnumerable<T> PreorderTraversal<T>(this T tree)
                 }
             }
         }
+
+        /// <summary>
+        /// Builds a copy of this tree with the same structure, but with different node values, as computed by the mapper argument.
+        /// mapper is called bottom-up, in reverse preorder traversal (i.e. children before the node, and the last child first before the first).
+        /// </summary>
+        [Pure]
+        public static Tree<TR> SelectNodeValue<TTree, T, TR>(this IRecursiveStructure<TTree, T> tree, Func<T, TR> mapper)
+            where TTree : IRecursiveStructure<TTree, T>
+            => tree.TypedThis.Select(node => mapper(node.NodeValue));
+
+        /// <summary>
+        /// Builds a copy of this tree with the same structure, but with different node values, as computed by the mapper argument.
+        /// mapper is called bottom-up, in reverse preorder traversal (i.e. children before the node, and the last child first before the first).
+        /// </summary>
+        [Pure]
+        public static Tree<TR> Select<TTree, TR>(this TTree tree, Func<TTree, TR> mapper)

```

Comment:
Deze werkt nu ook voor RootedTree - fijn, maar dat geeft soms ambiguity in de overload resolution.

Ik heb ervoor gekozen dus de variatie die de projectie doet obv de node value (dus zonder de node zelf) SelectNodeValue te noemen.  Maar ik ben niet 100% blij met die naam, en ik kan ook best leven met een minder linq-igy naam als `Map` oid.  suggesties welkom iig...